### PR TITLE
Coopermoore/sc 25/set task due dates

### PR DESCRIPTION
**Pull request by CooperMoore9**
Can set dues for tasks, also fixed a bug where you can enter a space for the task description and title and lose the ability to change them.

### CODE REVIEW

File: src/uiUpdater.ts [hunk 1]
```diff
@@ -170,21 +173,38 @@ export function loopTasks(selectedProject: Project) {
...
     }
+        
+    function dateChange(task: Task, project: Project) {
+        let taskDate = document.querySelector(`.taskDueDate${task.id}`)
+        taskDate?.replaceWith(document.createElement('input'))
+        let inputValue = document.querySelector('input')
+        inputValue?.setAttribute('type', 'date')
+        inputValue?.classList.add('h-7')
+        inputValue?.addEventListener('change', function(){
+            if(inputValue?.value)
```
Comment (Crowe7) [inline src/uiUpdater.ts:203]: does this if do anything? usually they would have `{}` after the statement

File: src/uiUpdater.ts [hunk 2]
```diff
@@ -196,9 +216,13 @@ export function loopTasks(selectedProject: Project) {
 
         inputValue?.addEventListener('keypress', function(event){
             if(event.key === 'Enter'){
-                if(inputValue?.value)
+                if(inputValue?.value && inputValue.value.trim() !== ''){
                 task.description = inputValue?.value
                 loopTasks(project)
+                }else{
+                    task.description = task.description
```
Comment (Crowe7) [inline src/uiUpdater.ts:223]: what does this do?

File: src/uiUpdater.ts (hunk 1 above)
Comment (CooperMoore9) [reply src/uiUpdater.ts:203]: typescript

**Review by Crowe7 — APPROVED** [review]

File: src/uiUpdater.ts (hunk 2 above)
Comment (CooperMoore9) [reply src/uiUpdater.ts:223]: #also fixed a bug where you can enter a space for the task description and title and lose the ability to change them.

**Pull request merged**